jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBufferPool.java

```diff
@@ -55,4 +60,56 @@ static RetainableByteBufferPool findOrAdapt(Container container, ByteBufferPool
         }
         return retainableByteBufferPool;
     }
+
+    /**
+     * A Pool wrapper that collects sample statistics on the acquired
+     * buffers.
+     * @see SampleStatistic
+     */
+    @ManagedObject
+    class SampledPool implements RetainableByteBufferPool, Dumpable
+    {
+        private final RetainableByteBufferPool _pool;
+        private final SampleStatistic _heap = new SampleStatistic();
+        private final SampleStatistic _direct = new SampleStatistic();
```

Comment:
I think those two counters are going to be a bit meaningless at the pool layer. IMHO it would be a lot more informative to move them to the `Bucket` layer, maybe by creating a `DumpableBucket` class.

Reply:
Hmmm good test for extensibility to have SampledBuckets.    Will try....

Reply:
Actaully,   thinking about it, we can just run with a small factor and then we will effectively get a histogram of the acquired sizes.    So maybe this class is not needed after all.